met-web/src/components/engagement/dashboard/report/EngagementDashboard.tsx

```diff
@@ -1,92 +1,124 @@
-import React, { useContext } from 'react';
-import { Grid } from '@mui/material';
-import { useAppSelector } from 'hooks';
-import { Link } from 'react-router-dom';
-import { useNavigate } from 'react-router';
-import { ActionContext } from '../../view/ActionContext';
-import { EngagementBanner } from '../../view/EngagementBanner';
-import { PreviewBanner } from '../../view/PreviewBanner';
-import { SubmissionStatus } from 'constants/engagementStatus';
+import React, { useEffect, useState } from 'react';
+import { Grid, Link as MuiLink, Skeleton, Typography } from '@mui/material';
+import { Link, useNavigate, useParams } from 'react-router-dom';
 import { AppConfig } from 'config';
-import { PrimaryButton } from 'components/common';
+import { MetPaper, PrimaryButton } from 'components/common';
+import { ReportBanner } from './ReportBanner';
+import { createDefaultEngagement, Engagement } from 'models/engagement';
+import { SubmissionStatus } from 'constants/engagementStatus';
+import { getEngagement } from 'services/engagementService';
+import { getErrorMessage } from 'utils';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { useAppDispatch } from 'hooks';
+
+export type EngagementParams = {
+    engagementId: string;
+};
 
 export const EngagementDashboard = () => {
-    const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
-    const isPreview = isLoggedIn;
-    const { savedEngagement } = useContext(ActionContext);
+    const { engagementId } = useParams<EngagementParams>();
     const navigate = useNavigate();
+    const dispatch = useAppDispatch();
     const urlpath = AppConfig.redashDashboardUrl;
-    const isOpen = savedEngagement.submission_status === SubmissionStatus.Open;
 
-    const handleStartSurvey = () => {
-        if (!isPreview) {
-            return;
+    const [engagement, setEngagement] = useState<Engagement>(createDefaultEngagement());
+    const [isEngagementLoading, setEngagementLoading] = useState(true);
+
+    const validateEngagement = (engagementToValidate: Engagement) => {
+        const neverOpened = engagementToValidate?.submission_status < SubmissionStatus.Open;
```

Comment:
whats the logic here?
the less than comparison might deserve a comment to explain it..

Reply:
I think @tiagograf-AOT made a point somewhere that there might be statuses other than pending open close (like draft) and these pages shouldn't be accessible if status was pending or draft, so open should have an id greater than id of pending and draft. so if engagementToValidate?.submission_status less than Open (i.e. pending or draft) it shouldn't be displayed

Reply:
As for clarifying the logic, the neverOpened variable name clears it out no?

Reply:
an explicit status check is much better..
very risky if id has to change in backend for some reason..I wouldn't expect the id has a meaning in terms of the values..
say we have to introduce a new status which belongs to open case, the new id will be more than that..so this logic will fail ..

Reply:
We have removed the incrementation of the id for the status models. Every new status we add we have to manually give the id number. I ask to consider which code you prefer:

engagementToValidate?.submission_status === SubmissionStatus.Pending || engagementToValidate?.submission_status === SubmissionStatus.Draft || engagementToValidate?.submission_status === SubmissionStatus.Scheduled

or just simply engagementToValidate?.submission_status < SubmissionStatus.Open? After having sorted the ids properly on create

Reply:
engagementToValidate?.submission_status  in ( SubmissionStatus.PendingSubmissionStatus.Draft,SubmissionStatus.Scheduled)

Or I would create a constant list with a sensisble name IS_NOT_OPEN_STATUSES =  [SubmissionStatus.PendingSubmissionStatus.Draft,SubmissionStatus.Scheduled]  and check engagementToValidate?.submission_status in IS_NOT_OPEN_STATUSES

Reply:
Understood.

handled in https://github.com/bcgov/met-public/pull/448